packages/react-aria-components/src/utils.tsx

```diff
@@ -210,9 +210,9 @@ export function useContextProps<T, U extends SlotProps, E extends Element>(props
   return [mergedProps, mergedRef];
 }
 
-export function useSlot(): [RefCallback<Element>, boolean] {
+export function useSlot(initialState: boolean | (() => boolean) = true): [RefCallback<Element>, boolean] {
   // Assume we do have the slot in the initial render.
```

Comment:
```suggestion
  // Initial state is typically based on the parent having an aria-label or aria-labelledby.
  // If it does, this value should be false so that we don't update the state and cause a rerender when we go through the layoutEffect
```